### PR TITLE
[SPIKE] Simple data-load panel

### PR DESCRIPTION
**Pull request by mauroservienti**
@adamralph this is the simplest form of `data-load` panel that can be add. We could also tweak it to make the loading message _bindable_ from the controller so that in case of error we can show some text in the same panel without adding complex directives or other components.

### CODE REVIEW

File: exercises/01-composite-ui/before/Divergent.Frontend/app/branding/orders/ordersView.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<section>
-    Orders
+    <h3>Orders list</h3>
```
Comment (adamralph) [inline exercises/01-composite-ui/before/Divergent.Frontend/app/branding/orders/ordersView.html]: I think we can drop the word "list".

File: exercises/01-composite-ui/before/Divergent.Frontend/app/branding/orders/ordersView.html [hunk 2]
```diff
@@ -12,4 +12,8 @@
         </li>
     </ul>
 
+    <div ng-show="ctrl.loadingData" class="data-load">
+        Data is loading....
```
Comment (adamralph) [inline exercises/01-composite-ui/before/Divergent.Frontend/app/branding/orders/ordersView.html]: Perhaps we could have a spinner here instead (with "Loading" alt text)? E.g.

ajax-loader

File: exercises/01-composite-ui/before/Divergent.Frontend/app/modules/sales/orders/ordersController.js [hunk 3]
```diff
@@ -6,13 +6,19 @@
...
                 ctrl.orders = null;
 
-                ctrl.isBusy = backendCompositionService
+                backendCompositionService
                     .get('orders-list', { pageIndex: 0, pageSize: 10 })
                     .then(function (viewModel) {
                         ctrl.orders = viewModel.orders;
+                    })
+                    .catch(function (error) {
+                        $log.error('Ooops, something went wroing: ', error);
```
Comment (adamralph) [inline exercises/01-composite-ui/before/Divergent.Frontend/app/modules/sales/orders/ordersController.js]: I agree that we should do _something_ in the UI in this case. I'm not sure exactly what's involved in your "bindable" suggestion, but it _sounds_ good. 😉

**Comment by mauroservienti** [comment]
> This approach looks good. Any downsides?

Not that I can think of. I'll get back to this probably next week.

**Comment by mauroservienti** [comment]
I took another direction, now it looks like this:

image

**Comment by mauroservienti** [comment]
And in case of an error a red panel appears

File: exercises/01-composite-ui/before/Divergent.Frontend/app/branding/orders/ordersView.html [hunk 4]
```diff
@@ -1,5 +1,5 @@
-﻿<section>
-    Orders
+﻿<section cg-busy="ctrl.isLoading">
+    <h3>Orders list</h3>
```
Comment (adamralph) [inline exercises/01-composite-ui/before/Divergent.Frontend/app/branding/orders/ordersView.html]: Can we remove the word "list"?

Comment (adamralph) [inline exercises/01-composite-ui/before/Divergent.Frontend/app/branding/orders/ordersView.html]: Can we remove the word "list"?

File: exercises/01-composite-ui/before/Divergent.Frontend/assets/site.css [hunk 5]
```diff
@@ -10,3 +10,13 @@ body {
...
+
+.data-load {
+    color: #5e5e5e;
+    height: 50px;
+    padding: 15px 15px;
+    font-size: 18px;
+    line-height: 20px;
+    font-style: italic;
+    text-align: center;
+}
```
Comment (adamralph) [inline exercises/01-composite-ui/before/Divergent.Frontend/assets/site.css]: This class is now redundant and should be removed.

File: exercises/01-composite-ui/before/Divergent.Frontend/Divergent.Frontend.csproj [hunk 6]
```diff
@@ -117,7 +120,9 @@
   <ItemGroup>
     <Compile Include="Properties\AssemblyInfo.cs" />
   </ItemGroup>
-  <ItemGroup />
+  <ItemGroup>
+    <Folder Include="app\directives\" />
```
Comment (adamralph) [inline exercises/01-composite-ui/before/Divergent.Frontend/Divergent.Frontend.csproj]: Is this required? The folder is empty.

File: exercises/01-composite-ui/before/Divergent.Frontend/lib/angular-busy/angular-busy.min.css [hunk 7]
```diff
@@ -0,0 +1,308 @@
+﻿.cg-busy {
```
Comment (adamralph) [inline exercises/01-composite-ui/before/Divergent.Frontend/lib/angular-busy/angular-busy.min.css]: This file is named `min` but is not minified. Is it the wrong file?

**Comment by mauroservienti** [comment]
This PR was meant just to show how to add a data loading, I plan to close this one once we are in agreement as it seems we are, and create a new one to implement it on all the exercises.

**Comment by adamralph** [comment]
@mauroservienti OK, understood. I'll rename to a spike then.

And I'm 👍 to go ahead with this approach!

**Comment by mauroservienti** [comment]
Closing as replaced by #101

**Pull request closed without merging**